### PR TITLE
inline CreateConfigMap into the places where it is used

### LINKED ISSUE #11436: test/utils.go: Inline CreateConfigMap
**Is your feature request related to a problem? Please describe**:

The function `CreateConfigMap` in tests/utils.go could be inlined, since it does not provide much additional value. Also since this is test code, assertions should be used instead of panicking on errors.

**Describe the solution you'd like**:

Inline `CreateConfigMap` into the places where it is used.

### PR DESCRIPTION
**Pull request by anishbista60**
### Description
Inlined ```CreateConfigMap``` present in `tests/utils.go`  into the places where it is used .
### What this PR does
Before this PR:

The functions `CreateConfigMap` in `tests/utils.go` doesnot look so useful because these function being called again and again in various file.

After this PR:

 This function is removed from `tests/utils.go` and inlined into the place where they are used.

Fixes #11436

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by anishbista60** [comment]
@0xFelix  please look over this changes

**Comment by 0xFelix** [comment]
That looks good so far. Only one nit, could you put something into the body of the commit message please? Something similar to what you have written in the PR description. Thanks!

Let's see what CI says
/test all

**Comment by 0xFelix** [comment]
@anishbista60 You need to rebase this PR before it can be merged, thanks.

**Comment by anishbista60** [comment]
@0xFelix Except squash commit . Is here any extra changes to be done ?

**Comment by 0xFelix** [comment]
Same comment as on #11450

**Comment by anishbista60** [comment]
@0xFelix Done sir and thankyou for guiding me.

**Comment by 0xFelix** [comment]
Sure and same comment as on #11450

**Comment by anishbista60** [comment]
@0xFelix  over here also

File: tests/config_test.go
Comment (0xFelix) [inline tests/config_test.go:50]: Was this reordered by `make format`?

**Review by 0xFelix — COMMENTED** [review]
One comment, otherwise looks good to me.

Can you give it a look too @alicefr?

File: tests/config_test.go
Comment (anishbista60) [reply tests/config_test.go:50]: @0xFelix  no sorry i did mistake while resloving merge conflict . shall i interchanged the /lipod and /kubevirt package.

Comment (anishbista60) [reply tests/config_test.go]: @0xFelix  Done

**Comment by anishbista60** [comment]
@alicefr sure, please look over it .

**Comment by alicefr** [comment]
Looks good!
/approve

**Review by 0xFelix — APPROVED** [review]
Thanks!

/lgtm

**Comment by alicefr** [comment]
@anishbista60 please fix the build failures. The files should be modified after the compilation. See `pull-kubevirt-build` CI job

**Comment by anishbista60** [comment]
i think i did one big mistake , when i cloned the repo. `git status` was showing 
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-ar
        modified:   bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-cpp
        modified:   bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-gcc
        modified:   bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-gcov
        modified:   bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-ld
        modified:   bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-nm
        modified:   bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-objdump
        modified:   bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-strip
        
        These changes are automatically happen so that  i discard this changes 
        
        with `git checkout --` 
        That's why job are failing
        
        Now, Can i retrive these changes ? How to solve that problem ?

**Comment by 0xFelix** [comment]
Run `make format` and amend the changes to your commit. That should solve the issue.

**Comment by anishbista60** [comment]
@0xFelix  shall i  format it from root  or /bazel folder.

**Comment by anishbista60** [comment]
@0xFelix   I run the make fomat from /kubervirt directory.

**Comment by anishbista60** [comment]
@0xFelix i'm scared looking what's happening

**Comment by anishbista60** [comment]
@0xFelix could you please checkout this changes

File: bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-ar
Comment (alicefr) [inline bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-ar:1]: Please, remove all those changes

**Comment by alicefr** [comment]
/hold
@anishbista60 please remove all the changes to the aarch64 files

File: bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-ar
Comment (anishbista60) [reply bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-ar]: @alicefr  i have made changes on the recommdation of @0xFelix( Run make format and amend the changes to your commit. That should solve the issue.)

**Comment by anishbista60** [comment]
@alicefr  but the working directory is clean . Why jobs are failing?

**Comment by 0xFelix** [comment]
Your formatting is off. You need to run `make format` at the project's root. Then you need to amend the changes to your commit.

**Comment by anishbista60** [comment]
> Your formatting is off. You need to run `make format` at the project's root. Then you need to amend the changes to your commit.

Is this a cause for failing of job. 
for format shall i run for main or inline-02 branch?

**Comment by 0xFelix** [comment]
You need to format your changes in `inline-02`.

**Comment by anishbista60** [comment]
@0xFelix yeah done sir , Run the `make format` from /kubevirt inline-02 branch  and ammed the changes .please check it out

**Comment by anishbista60** [comment]
@0xFelix sorry for push  , got confused with  changes made by `make format` and finally got to know that it was made by`make format`  and again rebase to previous changes.

**Comment by anishbista60** [comment]
@0xFelix  which changes do i need to take ? got confused

File: tests/config_test.go
Comment (0xFelix) [inline tests/config_test.go:161]: I'd swap order of empty line in Expect, so Expect comes immediately after the assignment to `err`, and then a new line.

**Review by 0xFelix — COMMENTED** [review]
@anishbista60 You need to rebase your PR on main again, so it can be merged.

File: tests/config_test.go
Comment (anishbista60) [reply tests/config_test.go:163]: @0xFelix should make changes to all file or just in `config_test.go`

**Comment by victortoso** [comment]
@anishbista60 how is it going with this one? Please remember to remove the merge commit. You shouldn't have merge commits in the PR, just the commits you want to add in the `main` branch.

**Comment by anishbista60** [comment]
@victortoso yeah , going to Modify it

File: tests/config_test.go
Comment (victortoso) [inline tests/config_test.go:373]: Just random statistics:

```
toso@tapioca ~/s/k/kubevirt (main)> grep -rniI "NotTo.Have" tests/ | wc -l
270
toso@tapioca ~/s/k/kubevirt (main)> grep -rniI "ToNot.Have" tests/ | wc -l
2487
```

So, `ToNot` is used about 10 times more in `tests/` than `NotTo`.

**Review by victortoso — CHANGES_REQUESTED** [review]
*pull-kubevirt-build* fails:
```
 Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   tests/config_test.go
	modified:   tests/migration/migration.go
	modified:   tests/operator/operator.go
	modified:   tests/sysprep_test.go
no changes added to commit (use "git add" and/or "git commit -a") 
```

running `make format` shows:
```
On branch review-11469
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   tests/config_test.go
        modified:   tests/migration/migration.go
        modified:   tests/operator/operator.go
        modified:   tests/sysprep_test.go

no changes added to commit (use "git add" and/or "git commit -a")
```

Please fix the issues locally in your machine before asking for another review.

File: tests/config_test.go
Comment (anishbista60) [reply tests/config_test.go:374]: @victortoso Done sir

**Comment by anishbista60** [comment]
@0xFelix I check the log but unable to figure out . could please suggest how to troubleshoot? OR do i need to run `make generate`

**Comment by victortoso** [comment]
/test pull-kubevirt-e2e-k8s-1.29-sig-storage
/test pull-kubevirt-e2e-arm64

>  I check the log but unable to figure out 

Just some unrelated issues, it seems. In the `sig-storage` we have `admission webhook "datavolume-mutate.cdi.kubevirt.io" denied the request: etcdserver: request timed out`. `e2e-arm64` shows a `panic` due lack of memory.

Please have pacience, your PR will be reviewed and if anything else is pending, we'll let you know.

**Comment by anishbista60** [comment]
@victortoso if you don't mind , could you please re run `pull-kubevirt-check-tests-for-flake`

**Comment by anishbista60** [comment]
@0xFelix could you please re run `pull-kubevirt-check-tests-for-flakes`

**Comment by anishbista60** [comment]
@alicefr unable to figure out the reason behind it . Could you please suggest?

**Comment by anishbista60** [comment]
@0xFelix could you please test it ?

**Comment by 0xFelix** [comment]
Sure!

/test all

**Comment by anishbista60** [comment]
@0xFelix every thing looks good but why job are failing. could you please assist me  and re Run the failed case .

**Comment by 0xFelix** [comment]
@anishbista60 Looks like flakes. Retesting.

/test pull-kubevirt-conformance-arm64
/test pull-kubevirt-e2e-k8s-1.29-sig-storage
/test pull-kubevirt-check-tests-for-flakes

**Comment by anishbista60** [comment]
@0xFelix please remove the `hold` label.

File: tests/config_test.go
Comment (0xFelix) [inline tests/config_test.go:37]: nit: Would you mind removing this empty line?

**Review by 0xFelix — COMMENTED** [review]
/hold cancel

**Comment by anishbista60** [comment]
@0xFelix remove the extra space. Now , i hope flakes job will pass .

**Comment by anishbista60** [comment]
@0xFelix now it's enough Seeing `flakes ` getting failed again and again . It's code fault or testing issue .

**Comment by 0xFelix** [comment]
@anishbista60 Not your fault.

/test pull-kubevirt-check-tests-for-flakes

**Comment by anishbista60** [comment]
@0xFelix no way Again failed .

**Comment by 0xFelix** [comment]
Looks to me like the lane is running longer than its definition permits.

@dhiller Can you help with this?

**Comment by anishbista60** [comment]
@0xFelix looks this Issue is already merged on someone else PR  .

**Comment by dhiller** [comment]
> Looks to me like the lane is running longer than its definition permits.
> 
> @dhiller Can you help with this?

@0xFelix hm, I see that the lane is trying to run 750 test cases, among those migration and operator tests. This is an edge case where not only large test files containing many tests have been changed but also those tests take a long time to run.

Honestly, I would not want to extend the time this lane runs to more than it is allowed to now. Instead we should work to split up the test files that have many tests into segments so that the number of tests inside a test file becomes smaller.

This would in many cases help reduce the number of tests being run on the check-test-for-flakes lane. A positive side effect of this would be making it easier for people coming into the project understand what the tests do.

WDYT?

**Pull request closed without merging**
